src/Illuminate/Support/Sleep.php

```diff
@@ -116,6 +107,32 @@ public static function sleep($duration)
         return (new static($duration))->seconds();
     }
 
+    /**
+     * Sleep for the given duration. Replaces any previously defined duration.
+     *
+     * @param  \DateInterval|int|float  $duration
+     * @return $this
+     */
+    public function duration($duration)
```

Comment:
Seems that this method can (and should) be protected. Or I am missing something?

Reply:
The idea is that in a Macro (or Conditionable -- see other PR) you can replace the duration of the sleep from outside of the `$this` scope, without spawning a brand new Sleep instance (as doing so would destruct the other, right?).

Reply:
The macro can access protected methods - as the given closure is bound to the sleep instance. As such, regarding this PR only, the method should be protected.